nuclearpasta/react-native-drax
# v0.11.0

## Introduction
When developing anything, there are usually 3 parts of the process:

1. Make it work
2. Make it right
3. Make it fast

- This PR is a Proof of concept of #165  where we're almost finalising step 1.

The code is a total mess, lots of TypeScript errors, bad constant namings, experiments, etc. But it's something that's working!

**▶️ Video:** https://x.com/LunatiqueCoder/status/1721982083902394694?s=20

##  How to test
1. Git clone this branch.
2. Run `yarn link` in the terminal where you cloned the branch.
3. Run `yarn && yarn tsc -w` in the terminal where you cloned the branch.
4. Run `yarn link react-native-drax` in the terminal where your project is (to link to the cloned branch). 
 
> If you don't have an existing drax project, you can git clone [LunatiqueCoder/expo-drax-example](https://github.com/LunatiqueCoder/expo-drax-example)
5. Run `rm -rf node_modules && yarn` to install the linked branch.

6. Your `metro.config.js` should look like this: 


```js
// Learn more https://docs.expo.io/guides/customizing-metro
const { getDefaultConfig } = require('expo/metro-config');
const path = require('path');

/** @type {import('expo/metro-config').MetroConfig} */
const config = getDefaultConfig(__dirname);


config.resolver.disableHierarchicalLookup = true;
// 2. Let Metro know where to resolve packages and in what order
config.resolver.nodeModulesPaths = [
  // Other packages, like redux-toolkit, might require the following as well
  path.resolve(__dirname, 'node_modules/@reduxjs/toolkit/node_modules'),

  // Because we're disabling hierarchical lookup, we have to tell what node_modules to use
  path.resolve(__dirname, 'node_modules'),
];

config.resolver.enableGlobalPackages = true;
config.watchFolders = ['/Users/myuser/OpenSource/react-native-drax'];


module.exports = config;
```

7. Run `yarn start` in your project.

8. Run the app
